pkg/services/dashboardversion/dashverimpl/dashver.go

```diff
@@ -339,5 +340,17 @@ func getRestoreVersion(msg string) (int, error) {
 		return 0, err
 	}
 
-	return int(ver), nil
+	safeVer, err := safeInt64ToInt(ver)
```

Comment:
That's overkill. You can just use strconv.Atoi https://pkg.go.dev/strconv#Atoi instead of `ver, err := strconv.ParseInt(parts[1], 10, 64)`